[DYN-3710] Enable Show CodeBlockNode Lines Number

Astul-Betizagasti opened the pull request:
### Purpose

Add and enable a toggle on the Preferences modal to turn CodeBlockNode Lines Number on and off
![image](https://user-images.githubusercontent.com/66637541/119373852-45a94b00-bc8f-11eb-8c17-bcde7a0020f6.png)

![codeblocklines2](https://user-images.githubusercontent.com/66637541/119394257-59ad7680-bca8-11eb-9cc4-b29371985dc8.gif)

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

Aaron Tang (@QilongTang )

### FYIs

Roberto Tellez (@RobertGlobant20 )

mjkkirschner commented:
@Astul-Betizagasti - if you check Dynamo 2.10 - when this setting is disabled, existing codeblocks should be modified, you should not need to recreate them to have the line numbers become visible or invisible.

Astul-Betizagasti commented:
> @Astul-Betizagasti - if you check Dynamo 2.10 - when this setting is disabled, existing codeblocks should be modified, you should not need to recreate them to have the line numbers become visible or invisible.

I just added a commit to fix this issue and updated the gif on the description to reflect the right behavior

Astul-Betizagasti commented:
I found a regression regarding the Saved Changes label on the Preferences modal and included a fix on this PR.
Now that the PreferencesViewModel is being saved on DynamoViewModel, instead of creating a new one every time the model opens, the Saved Changes label and its corresponding tooltip must be cleared every time the model is opened

QilongTang commented on src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:
```diff
@@ -641,8 +656,6 @@ public PreferencesViewModel(DynamoViewModel dynamoViewModel)
             SavedChangesLabel = string.Empty;
             SavedChangesTooltip = string.Empty;
 
-            this.PropertyChanged += Model_PropertyChanged;
```
Any reason why we delete this line?

Astul-Betizagasti replied:
yes, it was duplicated on line 650.

Astul-Betizagasti replied:
yes, it was duplicated in line 650

QilongTang replied:
I see, thanks @Astul-Betizagasti
